### PR TITLE
Convert FWCore/Integration tests to catch2

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Converted tests not using a testing framework.

#### PR validation:

Code compiles and tests pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Integration/test/ProcessConfiguration_t.cpp [hunk 1]
```diff
@@ -7,48 +9,47 @@
...
-    assert(pc1 == pc1);
-  }
-  {
-    edm::ProcessConfiguration pc1;
-    edm::ProcessConfiguration pc2;
-    pc1.setParameterSetID(id);
-    pc2.setParameterSetID(id);
-    assert(pc1 == pc2);
+  SECTION("default initialize") {
+    SECTION("elf operator==") {
```
Comment (makortel) [inline FWCore/Integration/test/ProcessConfiguration_t.cpp:17]: Should this be
```suggestion
    SECTION("self operator==") {
```
?

**Comment by makortel** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**